rtc_exec_enclave/src/types.rs

```diff
@@ -0,0 +1,9 @@
+use std::boxed::Box;
+
+pub(crate) type CallReturnValue = Box<[u8]>;
+
+pub(crate) type ExecResult = core::result::Result<CallReturnValue, ()>;
+
+pub(crate) trait ExecModule {
+    unsafe fn call(&self, dataset_ptr: *const u8, dataset_len: usize) -> ExecResult;
```

Comment:
The raw pointers here might make sense to move to another trait/function. The reason for the raw pointer operations is to mimic a wasmi workflow.

This is my understanding of the workflow we will need to follow so far.

To provide the data that will be used by the WASM module in the future, we need to load the data into the WASM instance that will run a function on the data. We should then be able to pass in a pointer as a WASM value type (I64 or I32) as well as the length as a value type.